source/coap_connection_handler.c

```diff
@@ -266,7 +266,10 @@ static internal_socket_t *int_socket_create(uint16_t listen_port, bool use_ephem
 
         // Set socket option to receive packet info
         socket_setsockopt(this->socket, SOCKET_IPPROTO_IPV6, SOCKET_IPV6_RECVPKTINFO, &(const bool) {1}, sizeof(bool));
-
+        if (socket_interface_selection != -1) {
+            // Select socket interface if selection requested
+            socket_setsockopt(this->socket, SOCKET_IPPROTO_IPV6, SOCKET_INTERFACE_SELECT, &socket_interface_selection, sizeof(int8_t));
```

Comment:
Oh, and if no interface is set, and it's greater than link-local, it uses the main routing table to choose an interface. Which normally means sending on the link towards your default route, unless you've manually added some routes for multicast addresses.

(Such "multicast routes" would only ever be used to choose the outgoing interface for our own multicast packets - they're nothing to do with multicast forwarding).